src/google/protobuf/testing/googletest.cc

```diff
@@ -94,6 +94,10 @@ string TestSourceDir() {
 namespace {
 
 string GetTemporaryDirectoryName() {
+  char *from_environment = getenv("TEST_TMPDIR");
```

Comment:
I added the comment, but I'm not sure what you mean about making sure other projects follow this. Maybe you're missing that Bazel automatically sets that environment variable for every test it runs?